module/x/gravity/keeper/batch.go

```diff
@@ -59,6 +59,10 @@ func (k Keeper) BuildOutgoingTXBatch(
 	}
 	k.StoreBatch(ctx, batch)
 
+	// Get the checkpoint and store it as a legit past batch
+	checkpoint := batch.GetCheckpoint(k.GetGravityID(ctx))
+	k.SetPastEthSignatureCheckpoint(ctx, checkpoint)
```

Comment:
how is this related to the evidence?

Reply:
~~also, why is this overriding the previous checkpoint? shouldn't you keep every checkpoint in-store instead of a single one?~~

See comment below about using the checkpoint as the key

Reply:
See comment below about our use of a set to store checkpoints.